pkg/controllers/status/work_status_controller.go

```diff
@@ -99,26 +91,25 @@ func (c *WorkStatusController) Reconcile(ctx context.Context, req controllerrunt
 		return controllerruntime.Result{Requeue: true}, fmt.Errorf("cluster(%s) not ready", cluster.Name)
 	}
 
-	return c.buildResourceInformers(cluster, work)
-}
-
-// buildResourceInformers builds informer dynamically for managed resources in member cluster.
-// The created informer watches resource change and then sync to the relevant Work object.
-func (c *WorkStatusController) buildResourceInformers(cluster *clusterv1alpha1.Cluster, work *workv1alpha1.Work) (controllerruntime.Result, error) {
-	err := c.registerInformersAndStart(cluster, work)
+	err = c.MemberClusterInformer.BuildResourceInformers(cluster, work, c.eventHandler)
 	if err != nil {
-		klog.Errorf("Failed to register informer for Work %s/%s. Error: %v.", work.GetNamespace(), work.GetName(), err)
 		return controllerruntime.Result{Requeue: true}, err
 	}
+
 	return controllerruntime.Result{}, nil
 }
 
-// getEventHandler return callback function that knows how to handle events from the member cluster.
-func (c *WorkStatusController) getEventHandler() cache.ResourceEventHandler {
-	if c.eventHandler == nil {
-		c.eventHandler = fedinformer.NewHandlerOnAllEvents(c.worker.Enqueue)
+func (c *WorkStatusController) onAdd(obj interface{}) {
+	curObj := obj.(runtime.Object)
+	c.worker.Enqueue(curObj)
+}
+
+func (c *WorkStatusController) onUpdate(old, cur interface{}) {
+	// Still need to compare the whole object because the interpreter is able to take any part of object to reflect status.
+	if !reflect.DeepEqual(old, cur) {
```

Comment:
The `resourceVersion` is always unequal, so it seems like it's always `true` here, right?